src/ui/browser_tabs.c

```diff
@@ -285,22 +289,14 @@ on_htmlview_close_tab (gpointer object, gpointer user_data)
 static void
 browser_tabs_close_tab (tabInfo *tab)
 {
-	int	n = 0;
-	GList	*iter, *list;
-
-	/* Find the tab index that needs to be closed */
-	iter = list = gtk_container_get_children (GTK_CONTAINER (tabs->notebook));
-	while (iter) {
-		if (tab->widget == GTK_WIDGET (iter->data))
+	g_autoptr(GListModel) list = gtk_notebook_get_pages (tabs->notebook);
+	for (guint n = 0; n < g_list_model_get_n_items (list); n++) {
+		/* Find the tab index that needs to be closed */
+		if (tab->widget == GTK_WIDGET (g_list_model_get_item (list, n))) {
```

Comment:
Note: `g_list_model_get_item()` transfers ownership of the returned value to the caller, which means you're leaking the item, here.

You should use something like:

```suggestion
                g_autoptr(GtkWidget) widget = g_list_model_get_item (list, n);
		if (tab->widget == widget) {
```

Reply:
Thanks!

Once everything is rewritten I will check for leakage with Valgrind. But first one has to replace everything and add 100 different mistakes :-)